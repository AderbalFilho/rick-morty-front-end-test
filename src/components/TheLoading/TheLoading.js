import React from 'react';

import { GText } from '@ui';

import loading from '@assets/imgs/loading.png';

import './TheLoading.scss';

function TheLoading() {
  return (
    <div className="the-loading">
      <img
        src={loading}
        alt="Rick and Morty visiting an alien planet"
        className="the-loading__img"
      />
      <GText classes="the-loading__text">Loading</GText>
    </div>
  );
}

export default TheLoading;
