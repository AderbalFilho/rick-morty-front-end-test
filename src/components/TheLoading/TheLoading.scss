@import 'styles/mixins';

.the-loading {
  align-items: center;
  background-color: var(--modal-background-color);
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: center;
  min-width: 100%;
  overflow-y: hidden;
  position: fixed;
  z-index: 1;
}

.the-loading__img {
  @include base-card;

  height: 185px;
  transform: var(--loading-transform);
  width: 185px;
}

.the-loading__text {
  font-size: 1.125rem;
  line-height: 1.5625rem;
  margin-top: 36px;

  @media (min-width: 768px) {
    font-size: 1.5rem;
    line-height: 2.0625rem;
  }
}
