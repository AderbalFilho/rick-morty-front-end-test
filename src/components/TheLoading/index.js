import TheLoading from './TheLoading';

export default TheLoading;
