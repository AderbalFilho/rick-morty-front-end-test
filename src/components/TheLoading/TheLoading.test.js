import React from 'react';
import TestRenderer from 'react-test-renderer';

import TheLoading from './TheLoading';

describe('TheLoading', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<TheLoading />);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });
});
