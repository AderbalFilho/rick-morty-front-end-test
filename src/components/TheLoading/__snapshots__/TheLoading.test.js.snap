// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheLoading matches snapshot 1`] = `
<div
  className="the-loading"
>
  <img
    alt="Rick and Morty visiting an alien planet"
    className="the-loading__img"
    src=""
  />
  <p
    className="g-text g-text--white g-text--nunito the-loading__text"
  >
    Loading
  </p>
</div>
`;
